crates/task-impls/src/quorum_proposal.rs

```diff
@@ -0,0 +1,501 @@
+use std::{collections::HashMap, sync::Arc};
+
+use async_broadcast::{Receiver, Sender};
+use async_lock::RwLock;
+use hotshot_task::{
+    dependency::{AndDependency, EventDependency, OrDependency},
+    dependency_task::{DependencyTask, HandleDepOutput},
+};
+use hotshot_types::{
+    consensus::Consensus,
+    data::{QuorumProposal, ViewChangeEvidence},
+    event::Event,
+    message::Proposal,
+    simple_certificate::QuorumCertificate,
+    traits::{
+        block_contents::BlockHeader,
+        network::{ConnectedNetwork, ConsensusIntentEvent},
+        node_implementation::{ConsensusTime, NodeImplementation, NodeType},
+    },
+    vote::HasViewNumber,
+};
+
+#[cfg(async_executor_impl = "async-std")]
+use async_std::task::JoinHandle;
+#[cfg(async_executor_impl = "tokio")]
+use tokio::task::JoinHandle;
+use tracing::{debug, error, instrument};
+
+use crate::{
+    consensus::CommitmentAndMetadata,
+    events::HotShotEvent,
+    helpers::{broadcast_event, cancel_task},
+};
+
+/// Validate a quorum proposal.
+#[allow(clippy::needless_pass_by_value)]
+fn validate_quorum_proposal<TYPES: NodeType>(
+    _quorum_proposal: Proposal<TYPES, QuorumProposal<TYPES>>,
+    _event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+) -> bool {
+    true
+}
+
+/// Validate a quorum proposal.
+#[allow(clippy::needless_pass_by_value)]
+fn validate_view_change_evidence<TYPES: NodeType>(
+    _certificate: ViewChangeEvidence<TYPES>,
+    _event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+) -> bool {
+    true
+}
+
+/// Validate a quorum proposal.
+#[allow(clippy::needless_pass_by_value)]
+fn validate_quorum_certificate<TYPES: NodeType>(
+    _certificate: QuorumCertificate<TYPES>,
+    _event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+) -> bool {
+    true
+}
+
+/// Validate a quorum proposal.
+#[allow(clippy::needless_pass_by_value)]
+fn validate_payload_and_metadata<TYPES: NodeType>(
+    _payload_commitment_and_metadata: CommitmentAndMetadata<<TYPES as NodeType>::BlockPayload>,
+    _event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+) -> bool {
+    true
+}
+
+/// Proposal dependency types. These types represent events that precipitate a proposal.
+#[derive(PartialEq)]
+enum ProposalDependency {
```

Comment:
Not sure how descriptive these names needed to be.

Reply:
This isn't a public enum so I think it's fine.